lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -1316,6 +1316,86 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     return b.create<arith::XOrIOp>(loc, payloadArgs[0], allOnesVal);
   }
 
+  if (isa<AtenDequantizeTensorOp, AtenDequantizeSelfOp>(op)) {
+    auto value = payloadArgs[0];
+    auto qtensor = op->getOperand(0);
+    auto qtensorTy = qtensor.getType().cast<ValueTensorType>().getDtype();
+    auto makeQTensor =
+        qtensor.getDefiningOp<Aten_MakePerTensorQuantizedTensorOp>();
+    if (!makeQTensor) {
+      qtensor.dump();
+      op->emitError("dequantizing tensor of unknown scale / zeorpoint");
+      return nullptr;
+    }
+
+    if (torch_to_linalg::isUnsignedTorchType(qtensorTy)) {
+      value = b.create<arith::UIToFPOp>(loc, payloadArgs[1].getType(), value);
+    } else {
+      value = b.create<arith::SIToFPOp>(loc, payloadArgs[1].getType(), value);
+    }
+
+    Value zp = makeQTensor.getZeroPoint();
+    zp = converter->materializeTargetConversion(
+        b, loc, converter->convertType(zp.getType()),
+        makeQTensor.getZeroPoint());
+    zp = b.create<arith::SIToFPOp>(loc, value.getType(), zp);
+    value = b.create<arith::SubFOp>(loc, value, zp);
+
+    Value scale = makeQTensor.getScale();
+    scale = converter->materializeTargetConversion(
+        b, loc, converter->convertType(scale.getType()),
+        makeQTensor.getScale());
+    scale = b.create<arith::TruncFOp>(loc, value.getType(), scale);
```

Comment:
Is the scale expected to always be a larger bit width than the target of `dequantize_per_tensor`? Or could we have, say, `f32` and `f32`? You can use `convertScalarToDtype(b, loc, scale, value.getType())` to handle other cases if so.

Reply:
This is more of a quirk of `torch` number values. They are always assumed to be `64-bit`, so we should pretty much always expect them to be truncated.